odoo/addons/base/models/ir_module.py

```diff
@@ -475,6 +475,8 @@ def button_immediate_install(self):
         :rtype: dict[str, object]
         """
         _logger.info('User #%d triggered module installation', self.env.uid)
+        if not self.env.registry.ready or self.env.registry._init:
+            raise UserError(_('The method button_immediate_install cannot be called on init or non loaded registries. Please use button_install instead.'))
```

Comment:
Why didn't you add this into method `_button_immediate_function` instead?  You don't want any "immediate" module operation to occur while loading a registry, do you?